digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/RoadAddressDAO.scala

```diff
@@ -1214,12 +1214,18 @@ object RoadAddressDAO {
         ""
       }
 
-      val historyFilter = if (includeHistory)
-        "AND end_date is null"
-      else
-        ""
-      val query =
-        s"""
+    val historyFilter = if (includeHistory)
+      "AND end_date is null"
+    else
+      ""
+
+    val validToFilter = if (rejectInvalids)
```

Comment:
I think this "valid_to is null" check should be always in query. We should treat the rows as deleted if valid_to is set.